src/silo/preprocessing/preprocessor.cpp

```diff
@@ -0,0 +1,515 @@
+      alias_key = PangoLineageAliasLookup::readFromFile(pango_lineage_definition_filename.value());
+   }
+
+   buildPartitioningTable();
+
+   if (preprocessing_config.getNdjsonInputFilename().has_value()) {
+      std::string order_by_select =
+         ", " + database_config.schema.primary_key + " as " + database_config.schema.primary_key;
+      if (database_config.schema.date_to_sort_by.has_value()) {
+         order_by_select += ", " + database_config.schema.date_to_sort_by.value() + " as " +
+                            database_config.schema.date_to_sort_by.value();
+      }
+      std::string partition_by_where, partition_by_select;
+      if (database_config.schema.partition_by.has_value()) {
+         partition_by_select = "partition_key_to_partition.partition_id as partition_id";
+         partition_by_where = fmt::format(
+            "where (preprocessing_table.{0} = partition_key_to_partition.partition_key) or "
+            "(preprocessing_table.{0} is null and "
+            "partition_key_to_partition.partition_key is null)",
+            database_config.schema.partition_by.value()
+         );
+      } else {
+         partition_by_select = "0 as partition_id";
+         partition_by_where = "";
+      }
+
+      for (const auto& [seq_name, _] : reference_genomes.raw_nucleotide_sequences) {
+         auto return_code = preprocessing_db.query(fmt::format(
+            "create or replace view nuc_{0} as\n"
+            "select {1} as key, nuc_{0} as sequence,"
+            "{3}"
+            "{2} \n"
+            "from preprocessing_table, partition_key_to_partition "
+            "{4};",
+            seq_name,
+            database_config.schema.primary_key,
+            order_by_select,
+            partition_by_select,
+            partition_by_where
+         ));
+
+         if (return_code->HasError()) {
+            SPDLOG_ERROR(return_code->GetError());
+            throw PreprocessingException(return_code->GetError());
+         }
+      }
+
+      for (const auto& [seq_name, _] : reference_genomes.raw_aa_sequences) {
+         auto return_code = preprocessing_db.query(fmt::format(
+            "create or replace view gene_{0} as\n"
+            "select {1} as key, gene_{0} as sequence, "
+            "{3}\n"
+            "{2} \n"
+            "from preprocessing_table, partition_key_to_partition "
+            "{4};",
+            seq_name,
+            database_config.schema.primary_key,
+            order_by_select,
+            partition_by_select,
+            partition_by_where
+         ));
+
+         if (return_code->HasError()) {
+            SPDLOG_ERROR(return_code->GetError());
+            throw PreprocessingException(return_code->GetError());
+         }
+      }
+   } else {
+      std::string order_by_select = ", partitioned_metadata." + database_config.schema.primary_key +
+                                    " as " + database_config.schema.primary_key;
```

Comment:
Do we need this? If I see it correctly, we could simply include `raw.key as {1}` in the statement below. `where raw.key = partitioned_metadata.{1}` should make sure that it's correct. Or is there a difference that I don't see?

In any other case, we could also include `partitioned_metadata.{1} as {1}` in the statement. I think it would still be more readable than constructing part of the select statement here.